### ISSUE TITLE
Callouts with highlight color

### ISSUE THREAD

**Issue by programlich** [RyotaUshio/obsidian-pdf-plus#27]
Is there a way for passing the selected hightlight color to the callout which is created when pasting the selection to my annotation note? That way the colors would match in the pdf and the annotation note and it would be easier for me to get them visually in line. It would be great to have the automatically colored callouts because when highlighting, the colors have distinct meanings for me and it would make reading my annotation note even more convenient to have this logic also applied in the callouts.

Right now I already love the view of callouts holding the annotations and my thoughts. It is very similar to the zotero view and would become even better with the automatic coloring functionality.

**Comment by RyotaUshio** [comment]
Thanks, this is a great suggestion!

The way PDF++ is currently implemented does not allow [CSS snippets](https://help.obsidian.md/Editing+and+formatting/Callouts#Customize%20callouts) to access highlight colors, but it will be not that hard to modify the code so that we can do something like below.

1. Make a CSS snippet:
 
```css
.callout[data-callout="pdf"][data-callout-metadata="My color"] {
   --callout-color: var(--pdf-plus-my-color-rgb);
}
```

2. Define a custom copy format: 

```
> [!pdf|{{colorName}}] page {{pageLabel}}
> {{selection}}

```

3. Then, the copied links will look like:

```markdown
> [!pdf|My color] page 1
> Lorem ipsum

```
and this callout will be colored with "My color".

**Comment by RyotaUshio** [comment]
Hi @programlich!

It's now available in [0.27.6](https://github.com/RyotaUshio/obsidian-pdf-plus/releases/tag/0.27.6)!

See the updated [README](https://github.com/RyotaUshio/obsidian-pdf-plus?tab=readme-ov-file#callout-colors) for the details.

**Comment by programlich** [comment]
This is so incredibly cool! I just tested the new feature and after a moment of getting to understand the new logic I managed to get it to work. Once I had figured out that I had to create one css snipped per color it worked like a breeze. It'so so helpufl to have to callouts colored the same as the hightlights. Just awsome, thank you so much.

I got to say this again: You are really nailing it with this plugin! It's exactly the one feature that was missing for obsidian to make it the all-in-one solution for scientists. Can't thank you enough.
